src/main/resources/avro/bdg.avdl

```diff
@@ -704,12 +704,83 @@ record Genotype {
   union { null, int }     phaseQuality = null;
 }
 
-record VariantEffect {
-  union { null, string} hgvs = null;
-  union { null, string } referenceAminoAcid = null;
-  union { null, string } alternateAminoAcid = null;
-  union {null, string} geneId = null;
-  union {null, string} transcriptId = null;
+/**
+ Estimate of putative impact of a variant.
+ */
+enum Impact {
+  HIGH,
+  LOW,
+  MODERATE,
+  MODIFIER // no known effect
+}
+
+enum MessageCode {
+  E1,
+  E2,
+  W1,
+  W2,
+  W3,
+  W4,
+  W5,
+  I1,
+  I2,
+  I3
+}
+
+enum MessageType {
+  ERROR_CHROMOSOME_NOT_FOUND, // code E1
+  ERROR_OUT_OF_CHROMOSOME_RANGE, // E2
+  WARNING_REF_DOES_NOT_MATCH_GENOME, // W1
+  WARNING_SEQUENCE_NOT_AVAILABLE, // W2
+  WARNING_TRANSCRIPT_INCOMPLETE, // W3
+  WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS, // W4
+  WARNING_TRANSCRIPT_NO_START_CODON, // W5
+  INFO_REALIGN_3_PRIME, // I1
+  INFO_COMPOUND_ANNOTATION, // I2
+  INFO_NON_REFERENCE_ANNOTATION // I3
+}
+
+/**
+ Errors, warnings, or informative messages regarding variant annotation accuracy.
+ */
+record Message {
+  union { null, MessageCode } code = null;
+  union { null, MessageType } type = null;
+}
+
```

Comment:
Yeah I think this sucks.  In java I'd have something like
```java
enum Message {
  ERROR_CHROMOSOME_NOT_FOUND(E1, "Chromosome does not exist..."), ...

  private final String code;
  private final String description;
  private Message(String code, String description) { ... }

  public static Message valueOfCode(final String code) {
    // lookup by code
  }
}
```

Don't know how to do something similar in avro.